### PR TITLE
fix: add empty rubric validation

### PR DESCRIPTION
**Pull request by dyudyunov**
**TL;DR -** Add a validation to forbid saving ORA response with an empty prompt to avoid errors.

image

This PR reopens https://github.com/openedx/edx-ora2/pull/2092 

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

1. Open unit for the course
1. Choose Open Response block - any type except "Peer assessment only" and "Self-assessment to peer assessment"
1. Click on Edit, on the Prompt tab click on Remove, save changes

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @dyudyunov! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by dyudyunov** [comment]
Hi @pomegranited 

I created a new branch based on the master and applied [Taras' fixes](https://github.com/openedx/edx-ora2/pull/2092).
Here are some notes about translations (unresolved in the old PR).

I have the gettext version
```
gettext --version
gettext (GNU gettext-runtime) 0.21.1
Copyright (C) 1995-2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Written by Ulrich Drepper.
```

I use the `make check_translations_up_to_date` command as described in [this guide](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)

**Review by pomegranited — APPROVED** [review]
Thank you for resubmitting this change @dyudyunov ! It works great.

I'll merge this now in case you need to submit version bump PRs to edx-platform before they cut quince.2 tonight.

>  Here are some notes about translations (unresolved in the old PR).

Your translation changes on this PR look fine -- you've resolved [the linebreaks that were in the previous diff](https://github.com/openedx/edx-ora2/pull/2092/files#diff-9e27eef519feb93c085bd760a3dd1aa3dc4695faeed9d59cc2dd2fc0cc547e69L1078), that's all I needed.

- [x] I tested this on my devstack:
   * Added a new ORA2 component to my test course
   * Removed all "prompts" from the component and tried to Save:
     Got the new error as expected: "You must provide at least one prompt."
  * Added a new prompt, and Saved without error.
- [x] I read through the code
- [x] I checked for accessibility issues by using my keyboard only during testing
- [x] <strike>Includes documentation</strike> N/A
- [x] User-facing strings are extracted for translation

**Comment by openedx-webhooks** [comment]
@dyudyunov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/2173&entry.1484655318=https://github.com/dyudyunov&entry.752974735=2024-02-08+13:41&entry.1917517419=2024-02-09+00:31&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by dyudyunov** [comment]
@pomegranited 

> Your translation changes on this PR look fine -- you've resolved [the linebreaks that were in the previous diff](https://github.com/openedx/edx-ora2/pull/2092/files#diff-9e27eef519feb93c085bd760a3dd1aa3dc4695faeed9d59cc2dd2fc0cc547e69L1078), that's all I needed.

I think the difference is that Taras ran those commands from the host machine (I was able to reproduce something very similar this way) instead of running them in the LMS container with ORA installed from the local dir.
Maybe the [guide](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md?rgh-link-date=2024-02-08T13%3A44%3A23Z) should be updated with a more detailed how-to for translations, e.g. adding a link to the [local development guide](https://github.com/openedx/edx-ora2/blob/master/docs/developers_guide.rst#using-ora-with-docker-devstack) seems helpful.
